torch_geometric/nn/conv/message_passing.py

```diff
@@ -99,8 +101,12 @@ def __init__(self, aggr: Optional[Union[str, List[str]]] = "add",
             assert len(set(aggr) | AGGRS) == len(AGGRS)
             self.aggr: Optional[str] = None
             self.aggrs: List[str] = aggr
+        elif isinstance(aggr, Aggregation):
+            self.aggr: Optional[Aggregation] = aggr
```

Comment:
Use `aggr_resolver` to map `str` to `Aggregation`?

Reply:
Do we want to resolve all the `str` in all the cases to `Aggregation`? It seems the `str` is used when `edge_index` is `SparseTensor` and used in `__repr__`. For some convs that override the `aggregate ` will also raise errors if `self.aggr` is not `str`. Should we keep both `str` and `Aggregation` as attributes and use them accordingly?

Reply:
That's a good point. Let's name it `aggr_module` or something like that.

Reply:
Ok. That should work.